### PR TITLE
Update EFM conformance suite scripts

### PR DESCRIPTION
**Pull request by derekgengenbacher-wf**
#### Reason for change
- Scripts to run the EFM conformance suite needed improving
- EFM conformance suite needs to be consumable by future CI tests

#### Description of change
- Updated EFM bash scripts to be a simple python script.
- Added a CI test for the EFM conformance suite
- Fixed an issue in PackageManager where packagesConfig wasnt being initialized correctly after a close/reset call
- Fixed an issue in PluginManager where pluginConfig wasnt being initialized correctly after a close/reset call
- Fixed an issue in PluginManager where pluginConfig wasnt being reset correctly after a close call
- Added unit tests around the above fix for PackageManager
- Added unit tests around the above fixes for PluginManager

#### Steps to Test
- Run the CI tests with `pytest --disable-warnings tests/integration_tests`. Not all tests will pass, 24 should fail. Run will take about 3 hours.
- Run the testing script with `python -m scripts.conformance_suites.run_efm_conformance_suite`. Run will take about 2 hours and 45 minutes.

**review**:
@Arelle/arelle

### CODE REVIEW

File: arelle/PackageManager.py [hunk 1]
```diff
@@ -296,7 +296,7 @@ def init(cntlr, loadPackagesConfig=True):
             packagesConfigChanged = False
         except Exception:
             pass # on GAE no userAppDir, will always come here
-    if packagesConfig is None:
+    if packagesConfig is None or len(packagesConfig) == 0:
```
Comment (austinmatherne-wk) [inline arelle/PackageManager.py:299]: May as well just check truthiness.
`if not packagesConfig:`

**Review by austinmatherne-wk — DISMISSED** [review]

File: scripts/conformance_suites/run_efm_conformance_suite.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+BASE_ARGS = [
+    '--logCodeFilter', FILTER,
+    '--packages', '{}'.format(os.path.abspath(TAXONOMY_PACKAGE)),
+    '--plugins', PLUGIN,
+    '--testcaseResultsCaptureWarnings',
+    '--validate'
+]
+
+IXBRL_ARGS = [
+    '--disclosureSystem', 'efm-pramatic',
```
Comment (stevenbronson-wk) [inline scripts/conformance_suites/run_efm_conformance_suite.py:20]: pragmatic?

Comment (derekgengenbacher-wf) [reply scripts/conformance_suites/run_efm_conformance_suite.py:14]: There are varying levels of validation for EFM. You can see them all in this config [file](https://github.com/Arelle/Arelle/blob/master/arelle/plugin/validate/EFM/config.xml). The SEC validates at level `efm-pragmatic`, which we match in wf_arelle. `efm-all-years` is used for deprecated taxonomies which are no longer accepted by the SEC.

Comment (danaoredson-wf) [reply scripts/conformance_suites/run_efm_conformance_suite.py:14]: I think steven is pointing out this misspelling
```suggestion
    '--disclosureSystem', 'efm-pragmatic',
```

**Review by austinmatherne-wk — APPROVED** [review]

File: arelle/PluginManager.py [hunk 3]
```diff
@@ -84,6 +84,7 @@ def save(cntlr):
         pluginConfigChanged = False
     
 def close():  # close all loaded methods
+    pluginConfig.clear()
```
Comment (danaoredson-wf) [inline arelle/PluginManager.py:87]: I haven't looked too hard, but if pluginConfig can be `None` (line 45), can this line throw an exception?

**Review by sagesmith-wf — APPROVED** [review]

File: arelle/PluginManager.py (hunk 3 above)
Comment (danaoredson-wf) [reply arelle/PluginManager.py:87]: Disregard, looks to not be an issue.

**Review by stevenbronson-wk — APPROVED** [review]

**Review by hermfischer-wf — APPROVED** [review]

**Comment by stevenbronson-wk** [comment]
Tests passed except for the expected 24 failures.

**Pull request merged**